provisioning_templates/user_data/autoyast_default_user_data.erb

```diff
@@ -0,0 +1,38 @@
+<%#
+kind: user_data
+name: AutoYaST default user data
+oses:
+- OpenSUSE
+- SLES
+-%>
+<%
+  # safemode renderer does not support unary negation
+  pm_set = @host.puppetmaster.empty? ? false : true
+  puppet_enabled = pm_set || @host.param_true?('force-puppet')
+  salt_enabled = @host.params['salt_master'] ? true : false
+-%>
+#!/bin/bash
+
+<%# Cloud instances frequently have incorrect hosts data %>
+<%= snippet 'fix_hosts' %>
+
+<% if @host.provision_method == 'image' && !root_pass.empty? -%>
+# Install the root password
+echo 'root:<%= root_pass -%>' | /usr/sbin/chpasswd -e
+<% end -%>
+
+<%= snippet('remote_execution_ssh_keys') %>
+
+<% if puppet_enabled %>
+<% if @host.param_true?('enable-puppetlabs-pc1-repo') -%>
+<%= snippet 'puppetlabs_repo' %>
+<% end -%>
+<%= snippet 'puppet_setup' %>
+<% end -%>
+
+<% if salt_enabled %>
+<%= snippet 'saltstack_setup' %>
+<% end -%>
```

Comment:
no :heart: for chef? I think it could be the same as in kickstart template https://github.com/theforeman/community-templates/blob/develop/provisioning_templates/provision/kickstart_default.erb#L203-L205